packages/async-flow/src/async-flow.js

```diff
@@ -0,0 +1,417 @@
+            // data in this internal flow instance -- the host activationArgs
+            // and the host outcome vow kit.
+            const guestResultP = (async () =>
+              // async IFFE ensures guestResultP is a fresh promise
+              apply(func, guestThis, guestArgs))();
+
+            bijection.define(guestResultP, outcomeKit.vow);
+            // log is driven at first by guestAyncFunc interaction through the
+            // membrane with the host activationArgs. At the end of its first
+            // turn, it returns a promise for its eventual guest result.
+            // It then proceeds to interact with the host through the membrane
+            // in further turns by `await`ing (or otherwise registering)
+            // on host vows turned into guest promises, and by calling
+            // the guest presence of other host objects.
+            //
+            // `bijection.hasGuest(guestResultP)` can be false in a delayed
+            // guest - to - host settling from a previous run.
+            // In that case, the bijection was reset and all guest caps
+            // created in the previous run were unregistered,
+            // including `guestResultP`.
+            void E.when(
+              guestResultP,
+              gFulfillment => {
+                if (bijection.hasGuest(guestResultP)) {
+                  outcomeKit.resolver.resolve(
+                    membrane.guestToHost(gFulfillment),
+                  );
+                  admin.done();
+                }
+              },
+              guestReason => {
+                // The `guestResultP` might be a failure thrown by `panic`
+                // indicating a failure to replay. In that case, we must not
+                // settle the outcomeVow, since the outcome vow only represents
+                // the settled result of the async function itself.
+                // Fortunately, `panic` resets the bijection, again resulting
+                // in the `guestResultP` being absent from the bijection,
+                // so this leave the outcome vow unsettled, as it must.
+                if (bijection.hasGuest(guestResultP)) {
+                  outcomeKit.resolver.reject(membrane.guestToHost(guestReason));
+                  admin.done();
+                }
+              },
+            );
+          },
+          wake() {
+            const { state, facets } = this;
+            const { flow } = facets;
+            const eph = tmp.for(flow);
+
+            if (failures.has(flow)) {
+              throw failures.get(flow);
+            }
+            if (state.isDone) {
+              return;
+            }
+            if (eph.membrane) {
+              eph.membrane.wake();
+            } else {
+              flow.restart();
+            }
+          },
+          getOutcome() {
+            const { state, facets } = this;
+            const { outcomeKit } = state;
+            const { flow } = facets;
+
+            flow.wake();
```

Comment:
The intuition was lazy evaluation. If someone wants the outcome, then if we were asleep, we should wake up so we can give them the outcome if we can. However, since this is a low level API and the `wake` method is exposed, I took your suggestion. Gone.

Reply:
I did not follow the intuition: if the flow is sleeping, that means it's pending on some external input. waking it will simply change its state, but not affect the outcome.

Anyway, since it's gone, no need for any follow-up.